source/known-hook-events.ts

```diff
@@ -20,7 +22,7 @@ export type BeforeRequestHook = (options: NormalizedOptions) => void | Promise<v
 /**
 Called with normalized [request options](https://github.com/sindresorhus/got#options). Got will make no further changes to the request. This is especially useful when you want to avoid dead sites.
 */
-export type BeforeRedirectHook = (options: NormalizedOptions) => void | Promise<void>;
+export type BeforeRedirectHook = (options: NormalizedOptions, response: http.ServerResponse | ResponseLike) => void | Promise<void>;
```

Comment:
Seeing as the `resonselike` package is meant to imitate a real response, we can just use the real response type.
```suggestion
export type BeforeRedirectHook = (options: NormalizedOptions, response: http.ServerResponse) => void | Promise<void>;
```

Reply:
It's not `ServerResponse`. It's `IncomingMessage`.

Reply:
Redirects aren't being cached, are they? No need to accept `ResponseLike` here.

Reply:
Ah! Didn't realize #797 added a type assertion.

https://github.com/sindresorhus/got/blob/4f6d287bf3a1fe62b5fed7b6a29b7ad8b797d792/source/request-as-event-emitter.ts#L122

So the hook should accept a `Response` and be passed `typedResponse` instead of accepting a `ServerResponse | ResponseLike` and being passed `response`.